src/server/scripts/Northrend/AzjolNerub/Ahnkahet/boss_amanitar.cpp

```diff
@@ -19,218 +19,264 @@
+    { 360.1683f, -889.1515f, -76.74798f, 3.612832f  },
+    { 350.1828f, -907.7313f, -74.94678f, 5.044002f  },
+    { 340.6278f, -856.5973f, -74.23862f, 4.415683f  },
+    { 366.4849f, -859.7621f, -74.82679f, 1.500983f  },
+    { 359.1482f, -853.3346f, -74.47543f, 5.654867f  },
+    { 374.9992f, -879.0921f, -75.56115f, 1.867502f  },
+    { 339.5252f, -850.4612f, -74.45442f, 4.764749f  },
+    { 337.0534f, -864.002f,  -75.72749f, 4.642576f  },
+    { 398.2797f, -851.8694f, -68.84419f, 0.5759587f }
+};
 
-        struct boss_amanitarAI : public BossAI
-        {
-            boss_amanitarAI(Creature* creature) : BossAI(creature, DATA_AMANITAR) { }
+struct boss_amanitar : public BossAI
+{
+    boss_amanitar(Creature* creature) : BossAI(creature, DATA_AMANITAR) { }
 
-            void Reset() override
-            {
-                _Reset();
-                me->SetMeleeDamageSchool(SPELL_SCHOOL_NATURE);
-                me->ApplySpellImmune(0, IMMUNITY_SCHOOL, SPELL_SCHOOL_MASK_NATURE, true);
-            }
+    void EnterCombat(Unit* /*who*/) override
+    {
+        _EnterCombat();
+        events.ScheduleEvent(EVENT_ROOT, Seconds(5), Seconds(9));
+        events.ScheduleEvent(EVENT_BASH, Seconds(10), Seconds(14));
+        events.ScheduleEvent(EVENT_BOLT, Seconds(15), Seconds(20));
+        events.ScheduleEvent(EVENT_MINI, Seconds(12), Seconds(18));
+        events.ScheduleEvent(EVENT_SPAWN, Seconds(1));
+        events.ScheduleEvent(EVENT_RESPAWN, Seconds(40), Seconds(60));
+    }
 
-            void EnterCombat(Unit* /*who*/) override
-            {
-                _EnterCombat();
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        _EnterEvadeMode();
+        summons.DespawnAll();
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_MINI);
+        _DespawnAtEvade();
+    }
 
-                events.ScheduleEvent(EVENT_ROOT, urand(5, 9) * IN_MILLISECONDS);
-                events.ScheduleEvent(EVENT_BASH, urand(10, 14) * IN_MILLISECONDS);
-                events.ScheduleEvent(EVENT_BOLT, urand(15, 20) * IN_MILLISECONDS);
-                events.ScheduleEvent(EVENT_MINI, urand(12, 18) * IN_MILLISECONDS);
-                events.ScheduleEvent(EVENT_SPAWN, 5  * IN_MILLISECONDS);
-            }
+    void JustDied(Unit* /*killer*/) override
+    {
+        _JustDied();
+        DoCastAOE(SPELL_REMOVE_MUSHROOM_POWER);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_MINI);
+    }
 
-            void JustDied(Unit* /*killer*/) override
-            {
-                _JustDied();
-                instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_MINI);
-            }
+    void JustSummoned(Creature* summon) override
+    {
+        summons.Summon(summon);
+    }
 
-            void SpawnAdds()
-            {
-                uint8 u = 0;
+    void SummonedCreatureDies(Creature* summon, Unit* killer) override
+    {
+        _mushroomsDeque.emplace_back(summon->GetPosition());
```

Comment:
emplace_back is an in-place constructor, use push_back

http://www.cplusplus.com/reference/vector/vector/emplace_back/